### DISCUSSION TITLE
Why does animating this list of widgets take forever to start?

### DISCUSSION THREAD

**Discussion by tusharsadhwani** [Textualize/textual#158] [Q&A]
Here's a really simple code:

```python
from rich.text import Text
from textual.app import App
from textual.widgets import Static


class MyApp(App):
    async def on_load(self) -> None:
        self.lines = [Static(Text("a")) for _ in range(self.console.width)]

    async def on_mount(self) -> None:
        await self.view.dock(*self.lines, edge="left", size=1)
        for line in self.lines:
            line.animate("layout_offset_y", 10, duration=10, easing=lambda x: x)


def main() -> None:
    """Main function"""
    MyApp.run()
```

Running this on my machine, the first frame of movement takes 3-4 seconds. Is there a better way to do this?

https://user-images.githubusercontent.com/43412083/140558472-98bc472d-450e-4459-b066-f5f6235fb66d.mp4

**Comment by willmcgugan** [comment]
I suspect the problem is the system I have in place to reduce screen updates. With so many things updating it doesn't have the change to insert a screen redraw.

I'll take a look next pass on the animation system.
